src/mem/ruby/system/GPUCoalescer.cc

```diff
@@ -129,8 +137,16 @@ UncoalescedTable::updateResources()
             instPktsRemaining.erase(seq_num);
 
             // Release the token
-            DPRINTF(GPUCoalescer, "Returning token seqNum %d\n", seq_num);
-            coalescer->getGMTokenPort().sendTokens(1);
+            if (!RubySystem::getWarmupEnabled()
+                    && !RubySystem::getCooldownEnabled()) {
```

Comment:
For this and other places where we are checking if Ruby is in warmup, we should probably comment why the code is conditioned out

Reply:
I added a few comments wherever Ruby checks if it is in warmup or cooldown.